### ISSUE TITLE
New warnings in Kokkos_Profiling.cpp

### ISSUE THREAD

**Issue by dutkalex** [kokkos/kokkos#5143]
Hi! I recently switched to Kokkos 3.6 (from 3.4) and I noticed several new warnings during my builds. Most of them could be disabled by turning off deprecated code (`Kokkos_ENABLE_DEPRECATED_CODE_3=Off`), but 3 `-Wmissing-field-initializers` are still there (on lines 166, 181 and 184). It doesn't seem to be anything severe, but it's somewhat annoying because builds with `-Wall -Wextra -Wall` now fail.

**Comment by PhilMiller** [comment]
> With respect of building your own code downstream, you should treat Kokkos includes as system headers (include them with `-isystem`) in which case you won't get these warnings.

Does using CMake `find_package(Kokkos)` and `target_link_library` as we recommend in the documentation do that?

**Comment by dalg24** [comment]
> > With respect of building your own code downstream, you should treat Kokkos includes as system headers (include them with `-isystem`) in which case you won't get these warnings.
> 
> Does using CMake `find_package(Kokkos)` and `target_link_library` as we recommend in the documentation do that?

Yes, as far as I know.
ArborX requires to build without warnings with `-Wall -Wextra -pedantic` and does
https://github.com/arborx/ArborX/blob/bc550946dc8a999419f32df7bdf10b174669dfa1/CMakeLists.txt#L3-L11

**Comment by ajpowelsnl** [comment]
@dutkalex -- have you had a chance yet to test these suggestions?

**Comment by dutkalex** [comment]
No, not yet @ajpowelsnl. As I mentioned in my first message, it is more an annoyance than a real problem and it got overtaken by more urgent matters in the meantime.

**Comment by ajpowelsnl** [comment]
Hi @dutkalex -- please give us a shout if the proposed solution does not work for you, and we'll reopen the issue.
